usaspending_api/api_contracts/contracts/autocomplete/City.md

```diff
@@ -0,0 +1,34 @@
+FORMAT: 1A
+HOST: https://api.usaspending.gov
+
+# City
+
+This end point returns a list of cities for a given limit, country, search string, and optional state code.
+
+## Advanced Search City Autocomplete [/api/v2/autocomplete/city/]
+
+Returns a list of cities matching search criteria
+
+### Advanced Search City Autocomplete [POST]
+
++ Request (application/json)
+    + Attributes (object)
+        + search_text: `Springfield` (required, string)
+        + limit: 40 (required, number)
+        + filter (object)
+            + country_code: `USA` (required, string)
+            + scope: `recipient_location` (required, enum[string])
+                + `primary_place_of_performance`
+                + `recipient_location`
+            + state_code: `SC` (optional, string)
+
++ Response 200 (application/json)
+    + Attributes (object)
+        + count: 10 (required, number)
+        + results (required, array[AutocompleteCityResult], fixed-type)
+
+# Data Structures
+
+## AutocompleteCityResult (object)
++ city_name: `Springfield` (required, string)
++ state_code: `VA` (required, string)
```

Comment:
Should `state_code` be optional for results outside the US?

Reply:
@IsaacRay can you confirm?

Reply:
@MaxwellKendall  yes, state_code should be optional, and, more specifically, unless state_code has a non-empty value, it should not be passed at all, because it trips up the request validator if you send it with an empty string.

Reply:
@IsaacRay This is the response object.

I would prefer we always keep responses consistent. If there is no state code n the DB, the value will be `null`. This shouldn't be a problem for the frontend since the user needs to select a country before using the city search. And that behavior of disabling states has been in place for some time.